nrf52-hal-common/src/timer.rs

```diff
@@ -17,6 +17,11 @@ use void::{unreachable, Void};
 #[cfg(any(feature = "52832", feature = "52840"))]
 use crate::target::{TIMER3, TIMER4};
 
+use core::marker::PhantomData;
+
+pub struct OneShot;
+pub struct Periodic;
```

Comment:
These could've been empty `enum`s

Reply:
I'm pretty new to Rust. Out of curiosity, what are the benefits of empty `enum`s versus empty `struct`s?

Reply:
Empty enums can't be constructed, which makes it explicit that these types only exist as compile-time markers. It's a minor thing though, so don't worry about it.

Reply:
That makes sense, thanks!